ucb-rit/coldfront
# Add command for creating/setting billing IDs and underlying manager classes

Fixes #465

**Changes**
- Added a `billling_ids` management command.
    - Added initial subcommands:
        1. `create`: Create `BillingActivity` objects.
        2. `set`: Set `BillingActivity` objects for use as:
            - `project_default`: A `Project`'s default billing ID,
            - `recharge`: The billing ID to be used for the recharge usage fee for a `ProjectUser`, or
            - `user_account`: the billing ID to be used for the user account fee for a `User`.
    - By default, the validity of a billing ID is verified before creating/setting it. This may be bypassed using the `--ignore_invalid` flag.
    - Updates may be staged using the `--dry_run` flag.
    - Additional subcommands (e.g., `list`) may be added in the future.
- Added a set of manager classes that allow the billing ID for a particular database entity (e.g., `Project`, `ProjectUser`, `User`) to be retrieved/set using a common interface with getter/setter methods.
    - Made these classes directly importable from `coldfront.core.billing.utils`.
- Added query methods for getting/creating a `BillingActivity` from a full ID and testing whether a full ID is well-formed.
- Added tests for the manager classes and the management command.

**How to Test**
- Review the code changes and add comments as needed.
- Ensure that the test suite passes on Jenkins.
